src/tiledimage.js

```diff
@@ -1148,27 +1150,33 @@ function compareTiles( previousBest, tile ) {
     return previousBest;
 }
 
-function drawTiles( tiledImage, lastDrawn ){
+function drawTiles( tiledImage, lastDrawn ) {
     var i,
         tile,
         tileKey,
         viewer,
         viewport,
         position,
-        tileSource;
+        tileSource,
+        contextSaved = false;
 
-    var usedClip = false;
-    if (tiledImage._clip) {
+    if ( tiledImage._clip ) {
         tiledImage._drawer.saveContext();
+        contextSaved = true;
+
         var box = tiledImage.imageToViewportRectangle(tiledImage._clip, true);
-        var topLeft = tiledImage.viewport.pixelFromPoint(box.getTopLeft(), true);
-        var size = tiledImage.viewport.deltaPixelsFromPoints(box.getSize(), true);
-        box = new OpenSeadragon.Rect(topLeft.x * $.pixelDensityRatio,
-            topLeft.y * $.pixelDensityRatio,
-            size.x * $.pixelDensityRatio,
-            size.y * $.pixelDensityRatio);
-        tiledImage._drawer.setClip(box);
-        usedClip = true;
+        var clipRect = tiledImage._drawer.viewportToDrawerRectangle(box);
+        tiledImage._drawer.setClip(clipRect);
+    }
+
+    if ( tiledImage.placeholderFillStyle && lastDrawn.length === 0 ) {
+        if ( !contextSaved ) {
+            tiledImage._drawer.saveContext();
+            contextSaved = true;
+        }
+
+        var placeholderRect = tiledImage._drawer.viewportToDrawerRectangle(tiledImage.getBounds(true));
+        tiledImage._drawer.drawPlaceholder(placeholderRect, tiledImage.placeholderFillStyle);
```

Comment:
I'm realizing now that it seems a little awkward that drawPlaceholder has this side effect (of setting the fill style) that we have to deal with by saving the context here. Maybe it would be better for drawPlaceholder to encapsulate the saving and restoring of the context. Sorry I didn't think of this sooner. 

It's different with setClip, where its explicit purpose is to change the context, and we can't restore until we've done all the drawing for this function. In the case of drawPlaceholder it can be entirely encapsulated (and it's fine to nest save/restore calls).